liberum_core/src/swarm_runner/messages.rs

```diff
@@ -92,22 +118,22 @@ impl SwarmContext {
             // Stops the swarm and informs the node
             SwarmRunnerMessage::Kill => Ok(true),
 
-            // Publish a file to the network
-            SwarmRunnerMessage::PublishFile {
+            // Start providing a file in the network
+            SwarmRunnerMessage::ProvideFile {
                 id,
                 path,
                 response_sender,
             } => {
-                if self.behaviour.published.contains_key(&id) {
+                if self.behaviour.providing.contains_key(&id) {
                     info!(
                         node = self.node.name,
                         id = format!("{id:?}"),
-                        "File is already published"
+                        "File is already being provided"
                     );
                     return Ok(false);
                 }
-                // Add the file to the published list
-                self.behaviour.published.insert(
+                // Add the file to the providing list
+                self.behaviour.providing.insert(
                     id.clone(),
                     file_share::SharedResource::File { path: path.clone() },
                 );
```

Comment:
It is not part of this PR, but why do we want to send GetProviders while handling ProvideFile message?

Reply:
I don't know what do you mean. GetProviders is not called. The `providing` hashmap is just a crude implementation of some kind of file store. It has a record <ID, PATH> for every file the Node is providing.